pkg/cache/partition_info.go

```diff
@@ -715,55 +715,46 @@ func checkAndSetResource(resource *resources.Resource) string {
 	if resource != nil {
 		return strings.Trim(resource.String(), "map")
 	}
-	return ""
+	return "[]"
 }
 
-// TODO fix this:
-// should only return one element, only a root queue
-// remove hard coded values and unknown AbsUsedCapacity
-// map status to the draining flag
-func (pi *PartitionInfo) GetQueueInfos() []dao.QueueDAOInfo {
-	pi.RLock()
-	defer pi.RUnlock()
-
-	var queueInfos []dao.QueueDAOInfo
-
-	info := dao.QueueDAOInfo{}
-	info.QueueName = pi.Root.Name
-	info.Status = pi.Root.stateMachine.Current()
-	info.Capacities = dao.QueueCapacity{
-		Capacity:        checkAndSetResource(pi.Root.GetGuaranteedResource()),
-		MaxCapacity:     checkAndSetResource(pi.Root.GetMaxResource()),
-		UsedCapacity:    checkAndSetResource(pi.Root.GetAllocatedResource()),
-		AbsUsedCapacity: "20",
-	}
-	info.ChildQueues = GetChildQueueInfos(pi.Root)
-	queueInfos = append(queueInfos, info)
-
-	return queueInfos
-}
-
-// TODO fix this:
-// should only return one element, only a root queue
-// remove hard coded values and unknown AbsUsedCapacity
-// map status to the draining flag
-func GetChildQueueInfos(info *QueueInfo) []dao.QueueDAOInfo {
-	var infos []dao.QueueDAOInfo
-	for _, child := range info.children {
-		queue := dao.QueueDAOInfo{}
-		queue.QueueName = child.Name
-		queue.Status = child.stateMachine.Current()
-		queue.Capacities = dao.QueueCapacity{
-			Capacity:        checkAndSetResource(child.GetGuaranteedResource()),
-			MaxCapacity:     checkAndSetResource(child.GetMaxResource()),
-			UsedCapacity:    checkAndSetResource(child.GetAllocatedResource()),
-			AbsUsedCapacity: "20",
+func calculateAbsUsedCapacity(capacity *resources.Resource, usedCapacity *resources.Resource) *resources.Resource {
+	if capacity == nil {
+		return nil
+	}
+	if usedCapacity == nil {
```

Comment:
I don't think it is a good idea to handle zero and nil in the same way. In case of nil I would return an empty resource. In case of all zeros I would do the calculations and will result in maxInt for zero capacity and zero resource for zero usage.